docs/conceptual/More-about-how-ROCm-uses-PCIe-Atomics.rst

```diff
@@ -1,156 +1,80 @@
 
-ROCm is an extension of HSA platform architecture, so it shares the queuing model, memory model,
-signaling and synchronization protocols. Platform atomics are integral to perform queuing and
-signaling memory operations where there may be multiple-writers across CPU and GPU agents.
+Platform atomics in ROCm
+==============================
+Platform atomics enables the set of atomic operations that perform RMW actions across multiple processors, devices, and memory locations to execute synchronously without interruption. An atomic operation is a sequence of computing instructions executed as a single, indivisible unit. These instructions are completed in their entirety without any interruptions or are not executed. These operations support 32-bit and 64-bit address formats.
 
-The full list of HSA system architecture platform requirements are here:
-`HSA Sys Arch Features <http://hsafoundation.com/wp-content/uploads/2021/02/HSA-SysArch-1.2.pdf>`_.
+ROCm uses platform atomics:
 
-AMD ROCm Software uses the new PCI Express 3.0 (Peripheral Component Interconnect Express [PCIe]
-3.0) features for atomic read-modify-write transactions which extends inter-processor synchronization
-mechanisms to IO to support the defined set of HSA capabilities needed for queuing and signaling
-memory operations.
+* Update the HSA queue's ``read_dispatch_id``. The command processor on the GPU agent uses a 64-bit atomic add operation. It updates the packet ID it processed.
+* Update the HSA queue's ``write_dispatch_id``. The CPU and GPU agents use a 64-bit atomic add operation. It supports multi-writer queue insertions.
+* Update HSA Signals. A 64-bit atomic operation is used for CPU & GPU synchronization.
 
-The new PCIe atomic operations operate as completers for ``CAS`` (Compare and Swap), ``FetchADD``,
-``SWAP`` atomics. The atomic operations are initiated by the I/O device which support 32-bit, 64-bit and
-128-bit operand which target address have to be naturally aligned to operation sizes.
 
-For ROCm the Platform atomics are used in ROCm in the following ways:
+PCIe for atomic operations
+----------------------------
 
-  * Update HSA queue's read_dispatch_id: 64 bit atomic add used by the command processor on the
-    GPU agent to update the packet ID it processed.
-  * Update HSA queue's write_dispatch_id: 64 bit atomic add used by the CPU and GPU agent to
-    support multi-writer queue insertions.
-  * Update HSA Signals -- 64bit atomic ops are used for CPU & GPU synchronization.
+PCIe supports the ``CAS`` (Compare and Swap), ``FetchADD``, and ``SWAP`` atomic operations across multiple resource. These atomic operations are initiated by the I/O devices that support 32-bit, 64-bit, and 128-bit operands. Likewise, the target memory address where these atomic operations are performed should also be aligned to the size of the operand. This alignment ensures that the operations are performed efficiently and correctly without failure. 
```

Comment:
```suggestion
PCIe supports the ``CAS`` (Compare and Swap), ``FetchADD``, and ``SWAP`` atomic operations across multiple resources. These atomic operations are initiated by the I/O devices that support 32-bit, 64-bit, and 128-bit operands. Likewise, the target memory address where these atomic operations are performed should also be aligned to the size of the operand. This alignment ensures that the operations are performed efficiently and correctly without failure. 
```

Is it the devices that support the operands, or the operations. It's unclear.  From the next sentence I assume it's the devices. Maybe split the sentence for a bit more clarity?